docs/src/content/en/docs/workflows/pausing-execution.mdx

```diff
@@ -5,56 +5,91 @@ description: "Pausing execution in Mastra workflows allows you to pause executio
 
 # Sleep & Events
 
-Pausing execution in Mastra workflows allows you to pause execution while waiting for external input or resources via `sleep()`, `sleepUntil()` and `waitForEvent()`.
+Mastra lets you pause workflow execution when waiting for external input or timing conditions. This can be useful for things like polling, delayed retries, or waiting on user actions.
 
-This sets the workflow status to `waiting`.
+You can pause execution using:
 
-## sleep()
+- `sleep()`: Pause for a set number of milliseconds
+- `sleepUntil()`: Pause until a specific timestamp
+- `waitForEvent()`: Pause until an external event is received
+- `sendEvent()`: Send an event to resume a waiting workflow
 
-`sleep()` pauses execution for a specified number of milliseconds.
+When using any of these methods, the workflow status is set to `waiting` until execution resumes.
 
-```typescript
-workflow
-    .then(step1)
-    .sleep(1000)
-    .then(step2)
-    .commit();
+## sleep
+
+The `sleep()` method pauses execution between steps for a specified number of milliseconds.
+
+```typescript {9} filename="src/mastra/workflows/test-workflow.ts" showLineNumbers copy
+import { createWorkflow, createStep } from "@mastra/core/workflows";
+import { z } from "zod";
+
+const step1 = createStep({...});
+const step2 = createStep({...});
+
+export const testWorkflow = createWorkflow({...})
+  .then(step1)
+  .sleep(1000)
+  .then(step2)
+  .commit();
 ```
 
-## sleepUntil()
+## sleepUntil
 
-`sleepUntil()` pauses execution until a specified date.
+The `sleepUntil()` method pauses execution between steps until a specified date.
 
-```typescript
-workflow
-    .then(step1)
-    .sleepUntil(new Date(Date.now() + 1000))
-    .then(step2)
-    .commit();
+```typescript {9} filename="src/mastra/workflows/test-workflow.ts" showLineNumbers copy
+import { createWorkflow, createStep } from "@mastra/core/workflows";
+import { z } from "zod";
+
+const step1 = createStep({...});
+const step2 = createStep({...});
+
+export const testWorkflow = createWorkflow({...})
+  .then(step1)
+  .sleepUntil(new Date(Date.now() + 5000))
+  .then(step2)
+  .commit();
 ```
 
-## waitForEvent()
+> `Date.now()` is evaluated when the workflow starts, not at the moment the `sleepUntil()` method is called.
```

Comment:
As discussed on Slack, will this be a callback in the next release?

Reply:
You can do both, but the callback is what I assume people will use pretty much always.